src/wrappers/themis/themispp/secure_comparator.hpp

```diff
@@ -33,34 +33,46 @@ namespace themispp{
       comparator_(NULL){
       res_.reserve(512);
       comparator_=secure_comparator_create();
-      if(!comparator_)
-	throw themispp::exception_t("Secure Comparator failed creating");
-      if(THEMIS_SUCCESS!=secure_comparator_append_secret(comparator_, &shared_secret[0], shared_secret.size()))
-	throw themispp::exception_t("Secure Comparator failed appending secret");	
+      if(!comparator_){
+        throw themispp::exception_t("Secure Comparator construction failed");
+      }
+      themis_status_t status=secure_comparator_append_secret(comparator_, &shared_secret[0], shared_secret.size());
+      if(THEMIS_SUCCESS!=status){
+        throw themispp::exception_t("Secure Comparator failed to append secret", status);
+      }
     }
 
     virtual ~secure_comparator_t(){
       secure_comparator_destroy(comparator_);
     }
 
     const data_t& init(){
+      themis_status_t status=THEMIS_FAIL;
       size_t data_length=0;
-      if(THEMIS_BUFFER_TOO_SMALL!=secure_comparator_begin_compare(comparator_, NULL, &data_length))
-	throw themispp::exception_t("Secure Comparator failed making initialisation message");
+      status=secure_comparator_begin_compare(comparator_, NULL, &data_length);
+      if(THEMIS_BUFFER_TOO_SMALL!=status){
+        throw themispp::exception_t("Secure Comparator failed to initialize comparison", status);
+      }
       res_.resize(data_length);
-      if(THEMIS_SCOMPARE_SEND_OUTPUT_TO_PEER!=secure_comparator_begin_compare(comparator_, &res_[0], &data_length))
-	throw themispp::exception_t("Secure Comparator failed making initialisation message");
+      status=secure_comparator_begin_compare(comparator_, &res_[0], &data_length);
+      if(THEMIS_SCOMPARE_SEND_OUTPUT_TO_PEER!=status){
+        throw themispp::exception_t("Secure Comparator failed to initialize comparison", status);
+      }
       return res_;
     }
 
     const data_t& proceed(const std::vector<uint8_t>& data){
+      themis_status_t status=THEMIS_FAIL;
```

Comment:
add check that data must be not empty?

Reply:
This one is actually important because `&data[0]` for empty `data` is technically undefined behavior.